Allow datatables to have conditional expanding rows

lennym opened the pull request:
Depending on a user's permissions and the data to be shown in a row, not all rows of the schedule of premises will have anything to display if they expand.

Make the expandable property of a datatable component conditional per-row to enable this.

lennym commented on src/filters/containers/connected-filters.js:
```diff
@@ -3,10 +3,10 @@ import { connect } from 'react-redux';
   key,
   title,
-  format,
+  format: formatFilterItems || format,
```
What's this bit here doing?

lennym replied:
I see, it allows a `formatFilterItems` method to exist on the schema and be used to format filter labels differently from formatting data table rows.

What's the use case for that, purely out of curiosity?

joefitter replied:
Yes - exactly that! Use case is the labels for holding/suitability codes in the datatable use the abbr tag with hover definition, but in the prototype, the labels for the filters show the definition then the abbreviation in brackets

lennym committed:
Allow datatables to have conditional expanding rows

Depending on a user's permissions and the data to be shown in a row, not all rows of the schedule of premises will have anything to display if they expand.

Make the expandable property of a datatable component conditional per-row to enable this.
